PublicAPI/Microsoft.ApplicationInsights.dll/net45/PublicAPI.Unshipped.txt

```diff
@@ -33,5 +33,237 @@ Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.IHeartbeatPro
+Microsoft.ApplicationInsights.Metrics.MetricIdentifier.MetricIdentifier(string metricNamespace, string metricId, string dimension1Name, string dimension2Name, string dimension3Name, string dimension4Name, string dimension5Name, string dimension6Name, string dimension7Name) -> void
+Microsoft.ApplicationInsights.Metrics.MetricIdentifier.MetricIdentifier(string metricNamespace, string metricId, string dimension1Name, string dimension2Name, string dimension3Name, string dimension4Name, string dimension5Name, string dimension6Name, string dimension7Name, string dimension8Name) -> void
+Microsoft.ApplicationInsights.Metrics.MetricIdentifier.MetricIdentifier(string metricNamespace, string metricId, string dimension1Name, string dimension2Name, string dimension3Name, string dimension4Name, string dimension5Name, string dimension6Name, string dimension7Name, string dimension8Name, string dimension9Name) -> void
+Microsoft.ApplicationInsights.Metrics.MetricIdentifier.MetricIdentifier(string metricNamespace, string metricId, string dimension1Name, string dimension2Name, string dimension3Name, string dimension4Name, string dimension5Name, string dimension6Name, string dimension7Name, string dimension8Name, string dimension9Name, string dimension10Name) -> void
+Microsoft.ApplicationInsights.Metrics.MetricIdentifier.MetricNamespace.get -> string
+Microsoft.ApplicationInsights.Metrics.MetricManager
+Microsoft.ApplicationInsights.Metrics.MetricManager.CreateNewSeries(Microsoft.ApplicationInsights.Metrics.MetricIdentifier metricIdentifier, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> dimensionNamesAndValues, Microsoft.ApplicationInsights.Metrics.IMetricSeriesConfiguration config) -> Microsoft.ApplicationInsights.Metrics.MetricSeries
+Microsoft.ApplicationInsights.Metrics.MetricManager.CreateNewSeries(string metricNamespace, string metricId, Microsoft.ApplicationInsights.Metrics.IMetricSeriesConfiguration config) -> Microsoft.ApplicationInsights.Metrics.MetricSeries
+Microsoft.ApplicationInsights.Metrics.MetricManager.CreateNewSeries(string metricNamespace, string metricId, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, string>> dimensionNamesAndValues, Microsoft.ApplicationInsights.Metrics.IMetricSeriesConfiguration config) -> Microsoft.ApplicationInsights.Metrics.MetricSeries
+Microsoft.ApplicationInsights.Metrics.MetricManager.Flush() -> void
+Microsoft.ApplicationInsights.Metrics.MetricManager.MetricManager(Microsoft.ApplicationInsights.Metrics.Extensibility.IMetricTelemetryPipeline telemetryPipeline) -> void
+Microsoft.ApplicationInsights.Metrics.MetricManager.Metrics.get -> Microsoft.ApplicationInsights.Metrics.MetricsCollection
+Microsoft.ApplicationInsights.Metrics.MetricSeries
```

Comment:
do we need to expose it? I haven't looked at it yet, but curious. I understand there is a `Metric` that you get via `GetMetric`, aggregator that you can configure for customization. What else do we need?

Reply:
I apologize, but I am not sure what you mean by "it" in your question "do we need to expose it?". 

The GitHub review tool more than 1 API to which this may refer. I will address your question to the best of my ability. Please let me know if you referred to a different API.

* The read-only `Metrics` property of the `Microsoft.ApplicationInsights.Metrics.MetricManager` class:

This provides access to all metrics in the manager's scope. A key use case is to allow users to manage existing metrics. For instance, consider an long-running application that manages containers with a limited life-time and tracks metrics about those containers. When a container life-time has finished, the business system can be sure that no further metrics about this container will be sent. However, the telemetry subsystem does not have this information. The `Metrics` collection allows a user to release no longer needed metrics and free up the corresponding memory.

* The `Microsoft.ApplicationInsights.Metrics.MetricSeries` class:

One of the primary requirements for the metrics APIs is to allow tracking 1000s of metric values per second with a small performance overhead. This requires the ability to obtain a handle to a specific metric series and to cache a reference to it so that it can be used to track values without the need for further look-ups. The `MetricSeries` class provides APIs for doing this without the need for any look-ups or object allocations.

This technique is discussed in detail in the samples 2 and 2a:
https://github.com/Microsoft/ApplicationInsights-dotnet/blob/feature/PreAggMetrics.Master/Test/Microsoft.ApplicationInsights.Test/Shared/Metrics/MetricsExamples.cs

 Please let me know if this answers your question.
If you require adjustments to these APIs, please request *precisely* the changes you would like to see so that I can make them.
Thank you.